### ISSUE TITLE
Composer 1.9.3 first installs, then downgrades

### ISSUE THREAD

**Issue by aschuch247** [composer/composer#8653]
My `composer.json`:

```json
{
    "name": "my/ci",
    "require-dev": {
        "php": "^5.5.0 || ^7.0.0",
        "mayflower/php-codebrowser": ">=1.1.0",
        "pdepend/pdepend": ">=2.5.0",
        "phploc/phploc": ">=2.1.0",
        "phpunit/dbunit": ">=2.0.0",
        "phpunit/phpunit": "^4.8.0 || ^5.7.0",
        "sebastian/phpcpd": ">=2.0.0"
    }
}
```

Output of `composer diagnose`:

```
Checking composer.json: WARNING
No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: FAIL
Missing pubkey for tags verification
Missing pubkey for dev verification
Run composer self-update --update-keys to set them up
Checking composer version: OK
Composer version: 1.9.3
PHP version: 7.2.23
PHP binary path: /home/aschuch/php/phpfarm-0.3.0/inst/php-7.2.23/bin/php

```

When I run this command:

```
$ ~/php/phpfarm-0.3.0/inst/bin/php-7.2.23 ~/php/composer/composer-1.9.3.phar update
```

I get the following output:

```
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 45 installs, 0 updates, 0 removals
  - Installing symfony/polyfill-ctype (v1.14.0): Loading from cache
  - Installing symfony/filesystem (v5.0.4): Loading from cache
  - Installing symfony/config (v5.0.4): Loading from cache
  - Installing psr/container (1.0.0): Downloading (100%)         
  - Installing symfony/service-contracts (v2.0.1): Downloading (100%)         
  - Installing symfony/contracts (v1.1.0): Downloading (100%)         
  - Installing symfony/dependency-injection (v4.2.8): Downloading (100%)         
  - Installing pdepend/pdepend (2.7.1): Loading from cache
  - Installing symfony/polyfill-php73 (v1.14.0): Downloading (100%)         
  - Installing symfony/polyfill-mbstring (v1.14.0): Loading from cache
  - Installing symfony/console (v4.4.4): Downloading (100%)         
  - Installing sebastian/version (2.0.1): Downloading (100%)         
  - Installing symfony/finder (v5.0.4): Loading from cache
  - Installing theseer/fdomdocument (1.6.6): Downloading (100%)         
  - Installing myclabs/deep-copy (1.9.5): Downloading (100%)         
  - Installing sebastian/resource-operations (1.0.0): Downloading (100%)         
  - Installing sebastian/recursion-context (2.0.0): Downloading (100%)         
  - Installing sebastian/object-enumerator (2.0.1): Downloading (100%)         
  - Installing sebastian/global-state (1.1.1): Downloading (100%)         
  - Installing sebastian/exporter (2.0.0): Downloading (100%)         
  - Installing sebastian/environment (2.0.0): Downloading (100%)         
  - Installing sebastian/diff (1.4.3): Downloading (100%)         
  - Installing sebastian/comparator (1.2.4): Downloading (100%)         
  - Installing symfony/yaml (v3.4.37): Loading from cache
  - Installing doctrine/instantiator (1.3.0): Loading from cache
  - Installing webmozart/assert (1.7.0): Loading from cache
  - Installing phpdocumentor/reflection-common (2.0.0): Loading from cache
  - Installing phpdocumentor/type-resolver (1.0.1): Loading from cache
  - Installing phpdocumentor/reflection-docblock (5.0.0): Loading from cache
  - Installing phpspec/prophecy (v1.10.2): Loading from cache
  - Installing phpunit/php-text-template (1.2.1): Downloading (100%)         
  - Installing phpunit/phpunit-mock-objects (3.4.4): Downloading (100%)         
  - Installing phpunit/php-timer (1.0.9): Loading from cache
  - Installing sebastian/code-unit-reverse-lookup (1.0.1): Downloading (100%)         
  - Installing phpunit/php-token-stream (2.0.2): Downloading (100%)         
  - Installing phpunit/php-file-iterator (1.4.5): Downloading (100%)         
  - Installing phpunit/php-code-coverage (4.0.8): Downloading (100%)         
  - Installing phpunit/phpunit (5.7.27): Downloading (100%)         
  - Installing sebastian/finder-facade (1.2.3): Loading from cache
  - Installing phploc/phploc (5.0.0): Downloading (100%)         
  - Installing phpunit/dbunit (2.0.3): Downloading (100%)         
  - Installing psr/log (1.1.2): Loading from cache
  - Installing monolog/monolog (1.25.3): Loading from cache
  - Installing mayflower/php-codebrowser (1.1.4): Downloading (100%)         
  - Installing sebastian/phpcpd (3.0.1): Downloading (100%)         
symfony/service-contracts suggests installing symfony/service-implementation
symfony/contracts suggests installing psr/cache (When using the Cache contracts)
symfony/contracts suggests installing symfony/cache-contracts-implementation
symfony/contracts suggests installing symfony/event-dispatcher-implementation
symfony/contracts suggests installing symfony/http-client-contracts-implementation
symfony/contracts suggests installing symfony/translation-contracts-implementation
symfony/dependency-injection suggests installing symfony/expression-language (For using expressions in service container configuration)
symfony/dependency-injection suggests installing symfony/proxy-manager-bridge (Generate service proxies to lazy load them)
symfony/console suggests installing symfony/event-dispatcher
symfony/console suggests installing symfony/lock
symfony/console suggests installing symfony/process
sebastian/global-state suggests installing ext-uopz (*)
phpunit/php-code-coverage suggests installing ext-xdebug (^2.5.1)
phpunit/phpunit suggests installing phpunit/php-invoker (~1.1)
phpunit/phpunit suggests installing ext-xdebug (*)
monolog/monolog suggests installing graylog2/gelf-php (Allow sending log messages to a GrayLog2 server)
monolog/monolog suggests installing sentry/sentry (Allow sending log messages to a Sentry server)
monolog/monolog suggests installing doctrine/couchdb (Allow sending log messages to a CouchDB server)
monolog/monolog suggests installing ruflin/elastica (Allow sending log messages to an Elastic Search server)
monolog/monolog suggests installing php-amqplib/php-amqplib (Allow sending log messages to an AMQP server using php-amqplib)
monolog/monolog suggests installing ext-amqp (Allow sending log messages to an AMQP server (1.0+ required))
monolog/monolog suggests installing ext-mongo (Allow sending log messages to a MongoDB server)
monolog/monolog suggests installing mongodb/mongodb (Allow sending log messages to a MongoDB server via PHP Driver)
monolog/monolog suggests installing aws/aws-sdk-php (Allow sending log messages to AWS services like DynamoDB)
monolog/monolog suggests installing rollbar/rollbar (Allow sending log messages to Rollbar)
monolog/monolog suggests installing php-console/php-console (Allow sending log messages to Google Chrome)
Package phpunit/phpunit-mock-objects is abandoned, you should avoid using it. No replacement was suggested.
Package phpunit/dbunit is abandoned, you should avoid using it. No replacement was suggested.
Writing lock file
Generating autoload files
```

When I run this command:

```
$ ~/php/phpfarm-0.3.0/inst/bin/php-7.2.23 ~/php/composer/composer-1.9.3.phar update
```

I get the following output:

```
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 2 installs, 14 updates, 6 removals
  - Removing symfony/polyfill-php73 (v1.14.0)
  - Removing symfony/contracts (v1.1.0)
  - Removing sebastian/resource-operations (1.0.0)
  - Removing sebastian/code-unit-reverse-lookup (1.0.1)
  - Removing myclabs/deep-copy (1.9.5)
  - Removing sebastian/object-enumerator (2.0.1)
  - Updating phpunit/php-timer (1.0.9 => 2.1.2): Loading from cache
  - Downgrading symfony/service-contracts (v2.0.1 => v1.1.8): Loading from cache
  - Installing symfony/debug (v4.4.4): Loading from cache
  - Downgrading symfony/console (v4.4.4 => v3.4.37): Loading from cache
  - Downgrading sebastian/version (2.0.1 => 1.0.6): Downloading (100%)         
  - Downgrading sebastian/environment (2.0.0 => 1.3.8): Downloading (100%)         
  - Downgrading sebastian/recursion-context (2.0.0 => 1.0.5): Downloading (100%)         
  - Downgrading sebastian/exporter (2.0.0 => 1.2.2): Downloading (100%)         
  - Downgrading phpunit/php-token-stream (2.0.2 => 1.4.12): Downloading (100%)         
  - Downgrading phpunit/phpunit-mock-objects (3.4.4 => 2.3.8): Downloading (100%)         
  - Downgrading phpunit/php-code-coverage (4.0.8 => 2.2.4): Downloading (100%)         
  - Updating sebastian/phpcpd (3.0.1 => 4.1.0): Loading from cache
  - Installing sebastian/git (2.1.4): Downloading (100%)         
  - Downgrading phploc/phploc (5.0.0 => 3.0.1): Loading from cache
  - Downgrading phpunit/phpunit (5.7.27 => 4.8.24): Loading from cache
  - Updating symfony/dependency-injection (v4.2.8 => v4.3.11): Loading from cache
Package phpunit/dbunit is abandoned, you should avoid using it. No replacement was suggested.
Package phpunit/phpunit-mock-objects is abandoned, you should avoid using it. No replacement was suggested.
Package sebastian/git is abandoned, you should avoid using it. No replacement was suggested.
Writing lock file
Generating autoload files
```

When I run this command:

```
$ ~/php/phpfarm-0.3.0/inst/bin/php-7.2.23 ~/php/composer/composer-1.9.3.phar update
```

I get the following output:

```
Loading composer repositories with package information
Updating dependencies (including require-dev)
Nothing to install or update
Package phpunit/dbunit is abandoned, you should avoid using it. No replacement was suggested.
Package phpunit/phpunit-mock-objects is abandoned, you should avoid using it. No replacement was suggested.
Package sebastian/git is abandoned, you should avoid using it. No replacement was suggested.
Generating autoload files
```

**reproduction**

* Remove `composer.lock`.
* Remove `vendor`.
* Run `composer update`.
* Run `composer update`.
* Run `composer update`.

**expectation**

I expect Composer to install the dependencies when run first. I expect Composer to not do anything any more when run again, provided that no update to the package registry happened in the meantime.

I do *not* expect to downgrade already installed packages.

**notes**

This issue might be remotely related to #8306, but there is no alternation between two states.

**Comment by Seldaek** [comment]
Probably fixed by 2.0 branch, but would be nice if you can confirm.

**Comment by Seldaek** [comment]
Indeed on 2.0 branch can not repro so closing this
